libraries/chain/controller.cpp

```diff
@@ -4446,14 +4447,15 @@ struct controller_impl {
             try {
                bool applied = apply_block( bsp, bsp->is_valid() ? controller::block_status::validated
                                                                 : controller::block_status::complete, trx_lookup );
+               --new_head_branch_size;
```

Comment:
Would `new_head_branch_size` ever be 0 before `--`?

Reply:
No